File: DSA/DevvratKhandal_590014906/Day_12/Question_1.java
import java.util.Scanner;

public class Question_1 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int[] width = new int[26];

        System.out.println("Enter widths of 26 letters:");
        for(int i = 0; i < 26; i++) {
            width[i] = sc.nextInt();
        }

        System.out.print("Enter the string: ");
        String s = sc.next();

        int lines = 1;
        int sum = 0;

        for(int i = 0; i < s.length(); i++) {

            int w = width[s.charAt(i) - 'a'];

            if(sum + w > 100) {
                lines++;
                sum = w;
            }
            else {
                sum += w;
            }
        }

        System.out.println("Total lines = " + lines);
        System.out.println("Width of last line = " + sum);
    }
}

File: DSA/Pulkit_Singh_590022181/Day12/Question1.c
#include <stdio.h>

int main() {
    int widths[26];

    for (int i = 0; i < 26; i++)
        scanf("%d", &widths[i]);

    char s[1001];
    scanf("%s", s);

    int lines = 1;
    int lastWidth = 0;

    for (int i = 0; s[i] != '\0'; i++) {
        int w = widths[s[i] - 'a'];

        if (lastWidth + w <= 100) {
            lastWidth += w;
        } else {
            lines++;
            lastWidth = w;
        }
    }

    printf("[%d, %d]", lines, lastWidth);

    return 0;
}